homeassistant/components/bloomsky/binary_sensor.py

```diff
@@ -1,39 +1,33 @@
 """Support the binary sensors of a BloomSky weather station."""
-import voluptuous as vol
 
 from homeassistant.components.binary_sensor import (
     DEVICE_CLASS_MOISTURE,
-    PLATFORM_SCHEMA,
     BinarySensorEntity,
 )
-from homeassistant.const import CONF_MONITORED_CONDITIONS
-import homeassistant.helpers.config_validation as cv
 
 from . import DOMAIN
 
-SENSOR_TYPES = {"Rain": DEVICE_CLASS_MOISTURE, "Night": None}
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-        )
-    }
-)
```

Comment:
I'm pretty sure this is going to break existing users who have `CONF_MONITORED_CONDITIONS` configured since it won't be a valid key anymore.

Reply:
This is correct, this makes changes to YAML configuration without introducing a configuration flow, which conflicts with ADR-0010.

A config flow should be introduced instead.

Reply:
Do either of you two have links/markdown/examples of how to integrate with the configuration flow? That ADR does not make it obvious.

Reply:
The ADR is merely a recording of the architectural decisions made for Home Assistant.
Information on development can be found in our developers documentation.

<https://developers.home-assistant.io/docs/creating_component_index>

Reply:
It is often unclear to me what the types and therefore what methods exist and/or return values are, so I just wind up opening a lot of files with vim. In particular with the `config` value that gets passed to `setup()`. 

Hmm, maybe http://dev-docs.home-assistant.io/en/master/api/config_entries.html is a decent place to find this information? Let me know if you have a better suggestion.

Reply:
API docs can be helpful, but won't tell a story.